model/project.go

```diff
@@ -564,42 +564,98 @@ var (
 	ProjectTasksKey         = bsonutil.MustHaveTag(Project{}, "Tasks")
 )
 
-func populateExpansions(d *distro.Distro, v *version.Version, bv *BuildVariant, t *task.Task, p *patch.Patch) *util.Expansions {
-	expansions := util.NewExpansions(map[string]string{})
+func PopulateExpansions(t *task.Task, h *host.Host) (util.Expansions, error) {
+	if t == nil {
+		return nil, errors.New("task cannot be nil")
+	}
+	if h == nil {
+		return nil, errors.New("host cannot be nil")
+	}
+	expansions := util.Expansions{}
 	expansions.Put("execution", fmt.Sprintf("%v", t.Execution))
 	expansions.Put("version_id", t.Version)
 	expansions.Put("task_id", t.Id)
 	expansions.Put("task_name", t.DisplayName)
 	expansions.Put("build_id", t.BuildId)
 	expansions.Put("build_variant", t.BuildVariant)
-	expansions.Put("workdir", d.WorkDir)
 	expansions.Put("revision", t.Revision)
 	expansions.Put("project", t.Project)
+
+	expansions.Put("workdir", h.Distro.WorkDir)
```

Comment:
I think you can delete this line. I think the `workdir` used to refer to the distro's working directory, but now it refers to the task's working directory. The agent overwrites this expansion in `startTask` with the expected value, so the value set here is never seen, and it's confusing to set it to something it doesn't actually refer to.

Reply:
This is actually also called from when we spawn hosts from tasks, but using it there is nonsensical so I removed it